File: JavaScript/1021 - Notas e Moedas.js
var N = parseFloat(require('fs').readFileSync('/dev/stdin', 'utf8'));
var notas = [100, 50, 20, 10, 5, 2];
var moedas = [1, 0.50, 0.25, 0.10, 0.05, 0.01];

console.log("NOTAS:");
for (let nota of notas) {
    let qtd = Math.floor(N / nota);
    console.log(`${qtd} nota(s) de R$ ${nota.toFixed(2)}`);
    N %= nota;
}

console.log("MOEDAS:");
for (let moeda of moedas) {
    let qtd = Math.floor(N / moeda + 0.00001);
    console.log(`${qtd} moeda(s) de R$ ${moeda.toFixed(2)}`);
    N %= moeda;
}

File: Clojure/1021 - Notas e Moedas.clj
(let [valor (Double/parseDouble (read-line))
      centavos-inicial (int (Math/round (* valor 100)))
      notas [10000 5000 2000 1000 500 200]
      moedas [100 50 25 10 5 1]]
  
  (println "NOTAS:")
  (let [resto-notas (reduce (fn [centavos nota]
                              (let [quantidade (quot centavos nota)]
                                (println (format "%d nota(s) de R$ %.2f" quantidade (/ nota 100.0)))
                                (mod centavos nota)))
                            centavos-inicial
                            notas)]

    (println "MOEDAS:")
    (reduce (fn [centavos moeda]
              (let [quantidade (quot centavos moeda)]
                (println (format "%d moeda(s) de R$ %.2f" quantidade (/ moeda 100.0)))
                (mod centavos moeda)))
            resto-notas
            moedas)))